src/plots/cartesian/axes.js

```diff
@@ -620,6 +620,15 @@ axes.calcTicks = function calcTicks(ax, opts) {
 
     generateTicks();
 
+    var isPeriod = ax.ticklabelmode === 'period';
+    if(isPeriod) {
+        // add one label to show pre tick0 period
+        tickVals = [{
+            minor: false,
+            value: axes.tickIncrement(tickVals[0].value, ax.dtick, !axrev, ax.caldendar)
+        }].concat(tickVals);
```

Comment:
not a big deal, but `tickVals.unshift({...})` is slightly more efficient here.

Reply:
Good call. 
I also fixed `reversed` range positioning and added few tests in https://github.com/plotly/plotly.js/pull/4993/commits/83f5cddbcbdac4f5c36f70f043a9156aaa1c0829.